File: org.palladiosimulator.blockchainsystems.core/src/main/java/org/palladiosimulator/blockchainsystems/core/transaction/TransactionStoredInPoolTraceEvent.java
package org.palladiosimulator.blockchainsystems.core.transaction;

import org.palladiosimulator.blockchainsystems.core.common.abstractions.TraceEvent;
import org.palladiosimulator.blockchainsystems.core.system.abstractions.Transaction;

public record TransactionStoredInPoolTraceEvent(
        long occurrenceTime,
        Transaction storedTransaction
) implements TraceEvent {

    public static final String EVENT_TYPE = "TransactionStoredInPoolTraceEvent";

    @Override
    public long getOccurrenceTime() {
        return occurrenceTime;
    }

    public Transaction getStoredTransaction() {
        return storedTransaction;
    }

    @Override
    public String getEventType() {
        return EVENT_TYPE;
    }

    @Override
    public void formatDetails(StringBuilder stringBuilder) {
        stringBuilder.append("{ txId=");
        stringBuilder.append(storedTransaction.getTxId());
        stringBuilder.append(" }");
    }
}

File: org.palladiosimulator.blockchainsystems.core/src/main/kotlin/org/palladiosimulator/blockchainsystems/core/transaction/TransactionStoredInPoolTraceEvent.kt
package org.palladiosimulator.blockchainsystems.core.transaction

import org.palladiosimulator.blockchainsystems.core.common.abstractions.TraceEvent
import org.palladiosimulator.blockchainsystems.core.system.abstractions.Transaction

/**
 * @author Davis Riedel
 */
@JvmRecord
data class TransactionStoredInPoolTraceEvent(
  private val occurrenceTime: Long,
  val storedTransaction: Transaction?
) : TraceEvent {
  companion object {
    const val EVENT_TYPE: String = "TransactionStoredInPoolTraceEvent"
  }

  override fun getOccurrenceTime(): Long = occurrenceTime
  override fun getEventType(): String = EVENT_TYPE

  override fun formatDetails(stringBuilder: StringBuilder) {
    stringBuilder.append("{ txId=")
    stringBuilder.append(storedTransaction!!.getTxId())
    stringBuilder.append(" }")
  }
}